### PR TITLE
docs(widget): typedoc for widget lib

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Described entities that needs an explanation

### CODE REVIEW

**Review by alfetopito — APPROVED** [review]

File: libs/widget-lib/src/JsonRpcManager.ts [hunk 1]
```diff
@@ -4,20 +4,36 @@ const JSON_PRC_V = '2.0'
 const TARGET_ORIGIN = '*'
 const EVENTS = ['connect', 'disconnect', 'close', 'chainChanged', 'accountsChanged']
 
+/**
+ * Manages JSON-RPC requests and interactions with an Ethereum provider.
+ */
 export class JsonRpcManager {
-  ethereumProvider: EthereumProvider | null = null
+  /** The Ethereum provider instance. */
```
Comment (anxolin) [inline libs/widget-lib/src/JsonRpcManager.ts:11]: should we mention what null means? it means the widget would handle the web3 connection, right?

File: libs/widget-lib/src/types.ts [hunk 2]
```diff
@@ -47,18 +80,43 @@ export interface CowSwapWidgetPalette {
...
   enabledTradeTypes?: TradeType[]
+  /**
+   * Colors palette to customize the widget UI.
+   */
   palette?: CowSwapWidgetPalette
 }
 
 export type CowSwapWidgetSettings = CowSwapWidgetUrlParams & CowSwapWidgetAppParams
 
 export interface CowSwapWidgetMetaData {
```
Comment (anxolin) [inline libs/widget-lib/src/types.ts:112]: there's some discussions about deleting he metadata, and moving appKey to the main object

**Review by anxolin — APPROVED** [review]

File: libs/widget-lib/src/JsonRpcManager.ts (hunk 1 above)
Comment (shoom3301) [reply libs/widget-lib/src/JsonRpcManager.ts:11]: Added a comment:
```
When is null the JSON-RPC manager is disconnected from the Ethereum provider
```

File: libs/widget-lib/src/types.ts (hunk 2 above)
Comment (shoom3301) [reply libs/widget-lib/src/types.ts:112]: Already done here:
https://github.com/cowprotocol/cowswap/pull/3318

**Pull request merged**